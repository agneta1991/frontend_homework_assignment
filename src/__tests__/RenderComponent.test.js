import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import DisplayData from '../components/DataFetching';

test('renders DisplayData component', () => {
  const { asFragment } = render(<DisplayData />);
  expect(asFragment()).toMatchSnapshot();
});

describe('DisplayData Component', () => {
  it('renders without crashing', () => {
    render(<DisplayData />);
  });

  it('displays loading message when loading is true', () => {
    render(<DisplayData />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });
});
