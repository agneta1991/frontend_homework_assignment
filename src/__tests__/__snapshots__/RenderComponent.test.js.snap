// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DisplayData component 1`] = `
<DocumentFragment>
  <div
    class="photo-div"
  >
    <div
      class="loading"
    >
      <p>
        Loading...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
